src/status.cc

```diff
@@ -250,3 +251,24 @@ void StatusPrinter::PrintStatus(Edge* edge, EdgeStatus status) {
                  force_full_command ? LinePrinter::FULL : LinePrinter::ELIDE);
 }
 
+void StatusPrinter::Warning(const char* msg, ...) {
```

Comment:
Hrmm...

Is using va_list really the only way?

I know we can't use C++11 parameter packs, but would it be practical to have an ifdef that uses a template parameter pack when C++11, and use va_list otherwise?

Reply:
#ifdef #else with two code paths isn't a good idea, because only one of them will be tested by most people.

Reply:
The real answer here is to adopt C++11.

My usage of Ninja will only ever use the template-parameter-pack implementation. Using va_lists is not type safe, and is prone to multiple issues that template-parameter-packs resolve.

Reply:
Justify your position please.

You are ignoring the standard mechanism to prevent bad behavior, which costs 2 lines of code, in favor of relying on arbitrary functionality offered by some compilers, some times, with no guarantee of working, and no guarantee that it'll continue working in future releases.

Reply:
???
I did:

> because only one of them will be tested by most people.

Reply:
I apologize. This is not the section that this comment was intended to be added to.

I don't understand how this happened.